pandas/core/generic.py

```diff
@@ -4287,6 +4287,37 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
         else:
             return result
 
+    def _clip_with_one_bound(self, threshold, method, axis, inplace):
+
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+
+        if np.any(isnull(threshold)):
+            raise ValueError("Cannot use an NA value as a clip threshold")
+
+        # method is self.le for upper bound and self.ge for lower bound
+        if is_scalar(threshold) and is_number(threshold):
+            if method.__name__ == 'le':
+                return self._clip_with_scalar(None, threshold, inplace=inplace)
+            return self._clip_with_scalar(threshold, None, inplace=inplace)
+
+        subset = method(threshold, axis=axis) | isnull(self)
+
+        # GH #15390
+        if is_scalar(threshold):
+            return self.where(subset, threshold, axis=axis, inplace=inplace)
+
+        # For arry_like threshold, convet it to Series with corret index
```

Comment:
as I said before, I don't think you need to do to *any* of this here.  simply pass ``np.asarray(threshold)`` to the ``self.where(...)``, it will broadcast correctly.